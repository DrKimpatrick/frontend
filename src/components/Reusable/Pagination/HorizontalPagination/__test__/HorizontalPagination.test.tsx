import React from 'react';
import ReactDOM from 'react-dom';
import Component from '../HorizontalPagination';
import { cleanup, render } from '@testing-library/react';

const props = {
  pageCount: 5,
  page: 0,
  onPageChange: jest.fn()
};

describe('HorizontalPagination', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Component {...props} />, div);
  });

  it('should create snapshot', () => {
    const tree = render(<Component {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
