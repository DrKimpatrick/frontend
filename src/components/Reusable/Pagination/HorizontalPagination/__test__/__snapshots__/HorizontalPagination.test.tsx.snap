// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalPagination should create snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="horizontalPagination"
      >
        <ul>
          <li
            class="previous disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              role="button"
              tabindex="0"
            >
              &lt;
            </a>
          </li>
          <li
            class="active"
          >
            <a
              aria-current="page"
              aria-label="Page 1 is your current page"
              role="button"
              tabindex="0"
            >
              1
            </a>
          </li>
          <li>
            <a
              aria-label="Page 2"
              role="button"
              tabindex="0"
            >
              2
            </a>
          </li>
          <li>
            <a
              aria-label="Page 3"
              role="button"
              tabindex="0"
            >
              3
            </a>
          </li>
          <li>
            <a
              aria-label="Page 4"
              role="button"
              tabindex="0"
            >
              4
            </a>
          </li>
          <li>
            <a
              aria-label="Page 5"
              role="button"
              tabindex="0"
            >
              5
            </a>
          </li>
          <li
            class="next"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              role="button"
              tabindex="0"
            >
              &gt;
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="horizontalPagination"
    >
      <ul>
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            role="button"
            tabindex="0"
          >
            &lt;
          </a>
        </li>
        <li
          class="active"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            role="button"
            tabindex="0"
          >
            1
          </a>
        </li>
        <li>
          <a
            aria-label="Page 2"
            role="button"
            tabindex="0"
          >
            2
          </a>
        </li>
        <li>
          <a
            aria-label="Page 3"
            role="button"
            tabindex="0"
          >
            3
          </a>
        </li>
        <li>
          <a
            aria-label="Page 4"
            role="button"
            tabindex="0"
          >
            4
          </a>
        </li>
        <li>
          <a
            aria-label="Page 5"
            role="button"
            tabindex="0"
          >
            5
          </a>
        </li>
        <li
          class="next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            role="button"
            tabindex="0"
          >
            &gt;
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
